### ISSUE TITLE
reopening bug

### ISSUE THREAD

**Issue by guycipher** [tidesdb/tidesdb#18]
There is a bug on reopening.  On reopening using `TidesDB::LSMT::New` the Get method is completed blocked.  I'm investigating why this is.  Essentially we should be reading sstable min and max and sstable pagers into memory.

**Comment by guycipher** [comment]
I'm investigating this currently but feel free to investigate and let me know what you find and propose for this.

**Comment by guycipher** [comment]
It seems to only be after reopening.  The sstables are populating fine, but operations are being blocked.

**Comment by guycipher** [comment]
The only thing happening in this case is when we have sstables to load, we load them, in descending order(modified by).  I'm wondering if it could be the Page locking mechanism here causing an issue.

**Comment by guycipher** [comment]
Further testing, it does not seem to be the pager after writing more pager tests.
I am going to further debug the reopen lsmt test and see what I get.

**Comment by guycipher** [comment]
![Screenshot from 2024-10-19 13-14-02](https://github.com/user-attachments/assets/269cd3b8-2466-4e20-b393-027e49c63aa5)
So as you can see we reopen, and on the first Get() operation it block forever waiting

Hm seems to be getting stuck in this loop for iteration
```
 for (const auto &sstable :
         std::vector<std::shared_ptr<SSTable>>(sstables.rbegin(), sstables.rend())) {
        if (!sstable) {
            continue;  // Skip null SSTables
        }

        // If the key is not within the range of this SSTable, skip it
        if (key < sstable->minKey || key > sstable->maxKey) {
            continue;
        }

        // Get an iterator for the SSTable file
        auto sstableIt = std::make_unique<SSTableIterator>(sstable->pager);
        if (!sstableIt) {
            continue;  // Skip if iterator creation failed
        }

        // Iterate over the SSTable
        while (sstableIt->Ok()) {
            auto kv = sstableIt->Next();
            if (!kv) {
                break;  // Break if no more key-value pairs
            }

            // Check for tombstones
            if (std::string(kv->value().begin(), kv->value().end()) == TOMBSTONE_VALUE) {
                return {};  // Return empty vector if tombstone is found
            }

            // Check for the key
            if (key ==
                ConvertToUint8Vector(std::vector<char>(kv->key().begin(), kv->key().end()))) {
                return ConvertToUint8Vector(
                    std::vector<char>(kv->value().begin(), kv->value().end()));
            }
        }
    }
```
In the get method, so what this is indicating is that on reopening preexisting sstables we are encountering an issue with the sstable iterator

The pager cant be the issue, I THINK as well, I wrote another pager test "pager2" under test directory and this test, reopens paged file no problem

**Comment by guycipher** [comment]
Commented out page locking within Pager::Read to test on re-open, its not the pager.

**Comment by guycipher** [comment]
I got it, it was the SSTable and SSTable iterator not using a proper pager pointer.

**Comment by guycipher** [comment]
Fixed in #21
